src/app/view/properties/still-image-osdviewer/still-image-osdviewer.component.ts

```diff
@@ -0,0 +1,112 @@
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * SALSAH is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * You should have received a copy of the GNU Affero General Public
+ * License along with SALSAH.  If not, see <http://www.gnu.org/licenses/>.
+ * */
+
+import {Component, Input, OnInit, OnChanges, SimpleChange, ElementRef} from '@angular/core';
+import {ReadStillImageFileValue} from "../../../model/webapi/knora/v2/read-property-item";
+
+// OpenSeadragon does not export itself as ES6/ECMA2015 module,
+// it is loaded globally in scripts tag of angular-cli.json,
+// we still need to declare the namespace to make TypeScript compiler happy.
+declare var OpenSeadragon: any;
+
+/**
+* This component creates a OpenSeadragon viewer instance.
+* Accepts an array of ReadStillImageFileValues to be rendered.
+* @class
+* @member {ReadStillImageFileValue[]} fileValues - fileValues to be rendered. (Use as angular @Input data binding property.)
+* @member {boolean} ignorePreviews - controls if entries in fileValues with isPreview==true should be ignored or not. (Use as angular @Input data binding property.)
+*/
+@Component({
+  selector: 'salsah-still-image-osdviewer',
+  templateUrl: './still-image-osdviewer.component.html',
+  styleUrls: ['./still-image-osdviewer.component.scss']
+})
+export class StillImageOSDViewerComponent implements OnInit {
+
+    @Input() fileValues: ReadStillImageFileValue[] = [];
+    @Input() ignorePreviews: boolean = true;
+
+    private viewer;
+
+    constructor(private elementRef: ElementRef) { }
+
+    ngOnInit() {
+      // initialisation is done on first run of ngOnChanges
+    }
+
+    ngOnChanges(changes:{[key: string]: SimpleChange}) {
+      if (changes["fileValues"].isFirstChange()) {
+        this.setupViewer();
+      }
+      this.openImages();
+    }
+
+    ngOnDestroy() {
+      if (this.viewer) {
+        this.viewer.destroy();
+        this.viewer = null;
+      }
+    }
+
+    private setupViewer() {
+      let viewerContainer = this.elementRef.nativeElement.getElementsByClassName("osdViewerContainer")[0];
+      let osdOptions = {
+        element: viewerContainer,
+        prefixUrl: "assets/icons/openseadragon/"
+      }
+      this.viewer = new OpenSeadragon.Viewer(osdOptions);
+      this.viewer.addHandler('full-screen', function (args) {
+        if (args.fullScreen) {
+          viewerContainer.classList.add("fullscreen");
+        } else {
+          viewerContainer.classList.remove("fullscreen");
+        }
+      })
+    }
+
+    private openImages() {
+      let imagesToDisplay = this.getImagesToDisplay();
+      let tileSources = [];
+      for (let i = 0; i < imagesToDisplay.length; i++) {
+        let sipiBasePath = imagesToDisplay[i].imageServerIIIFBaseURL + "/" + imagesToDisplay[i].imageFilename;
```

Comment:
I usually follow the policy 'implement the convenience method as soon as i have to write the same code again'. So far it's only used in one place. But i'm not opposed to having a convenience method